storage-dynamodb/src/main/java/zipkin2/storage/dynamodb/DynamoDBStorage.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2016-2019 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin2.storage.dynamodb;
+
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDBAsync;
+import java.math.BigInteger;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import zipkin2.internal.HexCodec;
+import zipkin2.storage.AutocompleteTags;
+import zipkin2.storage.SpanConsumer;
+import zipkin2.storage.SpanStore;
+import zipkin2.storage.StorageComponent;
+
+/**
+ * We use the following tables: zipkin-spans, zipkin-traces, zipkin-service-names,
+ * zipkin-span-names, zipkin-service-span-names
+ *
+ * zipkin-spans: trace_id span_id_composite (timestamp << 64 + (unsigned) span_id) span_id span
```

Comment:
Yeah, I've realized just now that I think I will do this differently. Dynamo will sort either numerically or alphabetically or on timestamp strings, depending on the type+shape. Hex sorted alphabetically is the same sort as numerically, I think it will be easier to reason about if I use a hex version of the timestamp + either span ID or a random long as hex, then I can just append 16 0s or Fs for the range query.

Reply:
So now, it's using the span timestamp with a random value appended, this gives us the uniqueness we need on the range key for inserts, and the range based queries/scans